src/main.rs

```diff
@@ -33,10 +36,25 @@ async fn main() {
         .get_info(tonic_lnd::lnrpc::GetInfoRequest {})
         .await
         .expect("failed to get info");
+    let info_clone = info.into_inner().clone();
+    let info_clone_2 = info_clone.clone();
 
-    let pubkey = PublicKey::from_str(&info.into_inner().identity_pubkey).unwrap();
+    let pubkey = PublicKey::from_str(&info_clone.identity_pubkey).unwrap();
     info!("Starting lndk for node: {pubkey}");
 
+    if !info_clone_2.features.contains_key(&ONION_MESSAGES_OPTIONAL) {
+        let mut node_info_retriever = GetInfoClient {
+            client: &mut client.lightning().clone(),
+        };
+        let mut announcement_updater = UpdateNodeAnnClient {
+            peers_client: client.peers(),
+        };
+        match set_feature_bit(&mut node_info_retriever, &mut announcement_updater).await {
+            Ok(_) => {}
+            Err(err) => panic!("Error setting feaure bit: {err}"),
```

Comment:
I've updated `main` to return `Result<()()>` in #20, can we duplicate that here? For the error case, we can `error!` log so that we can see the error string and then reutrn `Err()`